### ISSUE TITLE
code for an intelligent item-order in chest

### ISSUE THREAD

**Issue by loldermond** [kolton/d2bot-with-kolbot#2238]
i woke up this morning to this.
[https://www.directupload.net/file/d/5715/oyw6v95g_jpg.htm](url)
is there a way to prevent this mess aside from turning cubing entirely off?

**Comment by InterestingIndeed** [comment]
I don't think so. I think you just got really "unlucky" with the order of the drops. 

1. I would definitely recommend freeing up some space in your inventory! Running out of space prevents the bot from picking up items, which leads to no sold items = no money = no more pots/tps, bot stops running.

2. From my experience, you should have at least 4x4 free slots in the inventory. As you can see, the CA is filling up everything so the bot isnt able to pick up any more stuff (not even a ruby that could be cubed to free up space). If you would have additional 2x4 free slots in this situation, you bot could keep picking up small items and keep filling the chest. 

2. If you are cubing armors/weapons the bot might even find runes/gems to cube the armor/weapon, dropping them if they are wrong or have a chance to "reorder" after cubing. So there is a bit of fluctuation in the slots then.

3. I dont think you need so many skillers to farm hell. Maybe drop one, see how it goes, drop another. Im farming with a hammadin in a 3player game without any skillers, but having the same items as you do.

4. Whats with that book of identification blocking two slots?! ;)

5. If you dont want to / cant free up space, get a mule profile running. That will make sure you always have enough space.

**Comment by loldermond** [comment]
Whats with that book of identification blocking two slots?! ;)

-> saves time massively instead of buying a scroll, identifying, buying a scroll, identifying. worth it

**Comment by InterestingIndeed** [comment]
> 
> > Whats with that book of identification blocking two slots?! ;)
> 
> -> saves time massively instead of buying a scroll, identifying, buying a scroll, identifying. worth it
> 

how much time is actually saved? do you identify during the runs or only in town?

**Comment by DarkHorseDre** [comment]
Hey OP, if you havent solved yet, take a look at sonic (its on blizzhackers git which is also where kolton is being developed so move over there: https://github.com/blizzhackers/kolbot-sonic)

I dont know if the code is exposed yet as I haven't looked, but this was setup to reorganise the stash to achieve optimum space efficiency.
